### DISCUSSION TITLE
Auto grow the editor

### DISCUSSION THREAD

**Discussion by shashiOrdant** [ajaxorg/ace#5568] [Q&A]
Hi, I have been using "ace" 1..3.0, the issue I have is code in my editor is too long wich result into scroll.
What I am looking for is, I want the editor to have min height but on focus or click any such event the height should grow to content height.

**Comment by nightwing** [comment]
You can use combination of `maxLines` setting and css
```css
#myEditor {
    max-height: 5em;
    transition: max-height 0.5s
}
#myEditor.ace_focus {    
    max-height: 100em;    
}
```

See the demo [here](https://mkslanc.github.io/ace-playground/?value=Ly9zYW1wbGVzL2hlbGxvIHdvcmxkCgovLyBFYXNpZXN0IHdheSB0byBydW4gQUNFIGVkaXRvciBpbiBjb250YWluZXIgd2l0aCBpZD0iZXhhbXBsZSIKYWNlLmVkaXQoImV4YW1wbGUxIiwgewogICAgdGhlbWU6ICJhY2UvdGhlbWUvdGV4dG1hdGUiLAogICAgbW9kZTogImFjZS9tb2RlL2phdmFzY3JpcHQiLAogICAgdmFsdWU6ICIxXG4iLnJlcGVhdCg1MCksCiAgICBtYXhMaW5lczogMTUKfSk7CgphY2UuZWRpdCgiZXhhbXBsZTIiLCB7CiAgICB0aGVtZTogImFjZS90aGVtZS90ZXh0bWF0ZSIsCiAgICBtb2RlOiAiYWNlL21vZGUvamF2YXNjcmlwdCIsCiAgICB2YWx1ZTogIjFcbiIucmVwZWF0KDUwKSwKICAgIG1heExpbmVzOiAyMAp9KTtcMC5hY2VfZWRpdG9yIHsNCiAgICBtYXgtaGVpZ2h0OiA1ZW07DQogICAgdHJhbnNpdGlvbjogbWF4LWhlaWdodCAwLjVzDQp9DQouYWNlX2VkaXRvci5hY2VfZm9jdXMgeyAgICANCiAgICBtYXgtaGVpZ2h0OiAxMDBlbTsgICAgDQp9XDA8c2NyaXB0IHNyYz0iaHR0cHM6Ly93d3cudW5wa2cuY29tL2FjZS1idWlsZHNAbGF0ZXN0L3NyYy1ub2NvbmZsaWN0L2FjZS5qcyI%2BPC9zY3JpcHQ%2BCjxkaXYgaWQ9J2V4YW1wbGUxJyA%2BPC9kaXY%2BCjxicj4KPGJyPgo8YnI%2BCjxicj4KPGRpdiBpZD0nZXhhbXBsZTInID48L2Rpdj4%3D#hello-world)
